app/validation/i_validator.py

```diff
@@ -0,0 +1,8 @@
+from abc import ABCMeta, abstractmethod
+
+
```

Comment:
I don't like the file name for this class, none of the other "interface" classes have i_ in the file name.